clients/drcachesim/scheduler/scheduler_impl.cpp

```diff
@@ -564,8 +567,8 @@ scheduler_impl_tmpl_t<trace_entry_t, record_reader_t>::insert_switch_tid_pid(
     tid.size = 0;
     tid.addr = static_cast<addr_t>(input.tid);
 
-    input.queue.push_front(pid);
-    input.queue.push_front(tid);
+    input.queue.emplace_front(pid, IS_SYNTHETIC);
+    input.queue.emplace_front(tid, IS_SYNTHETIC);
```

Comment:
Hmm so this doesn't need to update `real_records_in_queue` b/c these are synthetic.

Maybe a helper is needed so adding goes through central places so no addition overlooks updating `real_records_in_queue`?  I'm afraid someone will see this code and add a real record to the queue here and not update the counter b/c the existing code doesn't update it.  Or just adding a comment here saying we don't need to increment is enough?